Add initial settings scaffolding

Regan-Koopmans opened the pull request:
@nickkossolapov I opened this PR mostly to initiate an async discussion. I cannot remember what the consensus was, but I think I recall some hesitancy about a modal-based settings page. If that is indeed the case then we can discuss some options here. These changes are a skeleton for what a settings page setup could be, barring styling and the actual settings themselves. 

The settings would probably be enough to justify their own slice, I will refactor the `show_settings` flag to that new slice if you agree with the general approach here.

nickkossolapov commented on src/components/shared/settings/Settings.tsx:
```diff
@@ -0,0 +1,25 @@
+
+export function Settings() {
+  const dispatch = useAppDispatch()
+  const show_settings = useAppSelector((state) => state.panel.show_settings)
```
Naming nit, but this should be camelCase. Maybe we should add a rule for this to Prettier or Eslint?

Regan-Koopmans replied:
My Python is showing, I can add a rule as part of this PR

nickkossolapov commented on src/app/viewSlice.ts:
```diff
@@ -3,10 +3,12 @@ import { Views } from "../domain/views"
 const initialState: ViewSlice = {
-  current: Views.MainFileSelector
+  current: Views.MainFileSelector,
+  show_settings: false
```
I think this is okay, and we can add a settings slice later. 

Also, I think for the settings we should also persist them to localstorage so they can last between sessions

Regan-Koopmans replied:
Good point, I'll work on that tonight

Regan-Koopmans commented:
I did a bit of refactoring, let me know what you think. It would be awesome to use the app in Chinese, so in anticipation for that I added a dropdown for interface language. This setting syncs with local storage. I added an option for Japanese too, since I figured this would be something you would be interested in. The actual translations will come in a separate PR, and would probably be the next thing I'll work on unless there is something else more critical.

nickkossolapov commented:
There's actually already a dropdown for language and it includes Chinese https://github.com/hyperbolae/shadowingly/blob/main/src/components/search/languageDropdown/LanguageDropdown.tsx

![image](https://github.com/user-attachments/assets/71c370b4-2182-406b-bdda-2b67ece3e6cd)

Maybe this can also set the settings and be loaded from it too?

Regan-Koopmans commented on src/domain/tatoeba.ts:
```diff
@@ -122,11 +122,11 @@ export function parseTatoebaSentence(sentence: TatoebaSentence): Sentence {
       id: sentence.audios[0].id,
       author: sentence.audios[0].author,
-      attribution_url: sentence.audios[0].attribution_url
+      attributionUrl: sentence.audios[0].attributionUrl
```
I added a prettier plugin to automatically update snake case to camel case. However in this case it would break the tatoeba integration right? I can maybe look into adding an opt-out for certain cases using a comment

nickkossolapov replied:
Yep, I think these would break it, as these are the JSON responses we get from Tatoeba

Regan-Koopmans replied:
Managed to find a way to do this

Regan-Koopmans commented on src/components/shared/settings/Settings.tsx:
```diff
@@ -0,0 +1,46 @@
+        id="language-select"
+        onChange={(event) => dispatch(setInterfaceLanguage(event.target.value))}
+      >
+        <option value="en">English</option>
```
So conceptually I think those languages are different. Someone might be Hungarian, and use the Hungarian localisation, but still choose Japanese Tataebo sound clips.

I understand your hesitation about non-native languages, I have removed Japanese as an option for now. I would still like to add Chinese in the near future, just because I think it would make the app look cool. But localisation will be a whole effort on its own, I just needed 1 setting to kick of the PR in the beginning.
